### DISCUSSION TITLE
Verifying minima in a growing list where the size is not fixed

### DISCUSSION THREAD

**Discussion by infinite-pursuits** [Consensys/gnark#947] [Q&A]
Hi, I have a list whose size keeps growing in an undetermined fashion. I need to check the minimum element in this list in my gnark circuit.
The issue I am facing is the size of the list.
```
type VerifyHeapPop struct {
	List    []frontend.Variable `gnark:"List"`
	MinItem frontend.Variable   `gnark:"MinItem"`
}
```

The above circuit gives the following error: 
```22:23:18 INF compiling circuit
ignoring uninitialized slice: List []frontend.Variable
22:23:18 INF parsed circuit inputs nbPublic=0 nbSecret=1
ignoring uninitialized slice: List []frontend.Variable
ignoring uninitialized slice: List []frontend.Variable
22:23:18 INF building constraint builder nbConstraints=0
22:23:18 WRN circuit has unconstrained inputs
panic: unrecognized R1CS curve type
```

If I instead define the circuit as below, the circuit takes too long as I need to run this circuit multiple times. Also max_possible_listsize comes from the user.

```
type VerifyHeapPop struct {
	List    [max_possible_listsize]frontend.Variable `gnark:"List"`
	MinItem frontend.Variable   `gnark:"MinItem"`
}

func (circuit *VerifyHeapPop) Define(api frontend.API) error {
	for j := 0; j < len(circuit.List); j++ {
		api.Println("jth element: ", j, circuit.List[j])
		api.AssertIsLessOrEqual(circuit.MinItem, circuit.List[j])
	}
	return nil
}
```

a) Can I somehow fix max_possible_listsize to a user fed input?
b) Is there some other way to find the minimum in a growing list where I dont know the size in advance?

**Comment by ivokub** [comment]
If the size of the slice is part of the witness then you have to pad the check inside the circuit. Usually my approach is to just pad with valid data (for example with variable `Fr-1`), so the check for padded input is always valid.
